src/NUnitTestExtractor/NUnitTestExtractor/Program.cs

```diff
@@ -32,68 +33,132 @@ static void Main(string[] args)
                     {
+                        StreamWriter writer =null;
                         try
                         {
                             level = (Level)Enum.Parse(typeof(Level), _options.Level, true);
-                            WriteTestDllAndName("file", _options.Output, _options.DLLs, level);
+
+                            if (_options.Output == null)
+                            {
+                                writer = new StreamWriter(Console.OpenStandardOutput());
+                            }
+                            else
+                            {
+                                writer = new StreamWriter(_options.Output, true);
+                            }
+
+                            GetTests(_options.DLLs, level, writer);
                         }
                         catch (ArgumentException e)
                         {
                             Console.Error.WriteLine(e.GetType() + ": Please enter either Namespace, Class, or Function for level");
                             Environment.Exit(-1);
                         }
+                        finally
+                        {
+                            if (writer != null)
+                            {
+                                writer.Dispose();
+                                
+                            }
+                        }
+
                     }
                 }
             }
         }
+
         /// <summary>
-        /// While reflection is not implemented method writes each dll verified to be existing to either a txt file or stdout along with the specified level
+        /// Writes each DLL file path along with either a namespace, class or function which contains tests(dependent on level specified by user)
         /// </summary>
-        /// <param name="name"> the name of the txt file that data is being appended to </param>
-        /// <param name="outputDirectory"> the directory in that the file is being saved to </param>
-        /// <param name="dllDirectories"> list of all dll directories which the user wishes to append to a file or stdout </param>
-        /// <param name="level"> Specifies the level of granuality to use for the output tests entered by user. </param>
-        private static void WriteTestDllAndName(string name, string outputDirectory, IList<string> dllDirectories, Level level)
+        /// <param name="writer"> StreamWriter used to write the dll path and data, either to a txt file or stdout </param>
+        /// <param name="dll"> The dll path which is written </param>
+        /// <param name="data"> The test information corrosponding to the dll file </param>
+        private static void WriteTestDllAndName(StreamWriter writer, string dll, string data)
         {
-            //user entered some dlls
-            if (dllDirectories.Count != 0)
-            {
-                StreamWriter writer;
+            writer.Write(dll + "|" + data + Environment.NewLine);
```

Comment:
You can use " | " instead of "|" to make the output a little more Human readable.
It's easier to see where the two parts are if there's some space before & after the pipe.